### PR TITLE
Feature/insect selec improv

### PR DESCRIPTION
**Pull request by bestrandomnameever**
Main contribution: enforce placing queen by 4th turn in UI.
Some other small fixes and improvements.

### CODE REVIEW

File: config/webrtc_ICE_servers.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[
```
Comment (WardMaes) [inline config/webrtc_ICE_servers.json:1]: This file is never used

File: components/InsectSelector.tsx [hunk 2]
```diff
@@ -25,27 +25,34 @@ const InsectSelector = ({ playerHand: insects }: InsectSelectorProps) => {
...
+          <div
+            className={classNames('insect', {
+              insect__selected: isSelectedInsect,
+              insect__disabled: !playerToMove || !insectCanBePlayed,
+            })}
+            key={i + 1}
+            onClick={() => {
+              ;(playerToMove || !insectCanBePlayed) &&
+                sendToGame('UNPLAYEDPIECE.SELECT', { insectName })
+            }}
```
Comment (WardMaes) [inline components/InsectSelector.tsx:44]: `if(playerToMove || !insectCanBePlayed){
  sendToGame
}`

File: components/InsectSelector.tsx [hunk 3]
```diff
@@ -25,27 +25,34 @@ const InsectSelector = ({ playerHand: insects }: InsectSelectorProps) => {
...
+          >
+            <Image
+              src={`/icons/${insectName.toLowerCase()}.svg`}
+              width="50"
+              height="50"
```
Comment (WardMaes) [inline components/InsectSelector.tsx:49]: Is this responsive?

File: lib/p2p.ts [hunk 4]
```diff
@@ -151,3 +103,35 @@ export async function sync(context: Context) {
...
+  return new Peer({
+    id: id,
+    host: 'peer-connection.herokuapp.com',
+    secure: true,
+    port: 443,
+    path: '/peerjs/hive',
+    initiator: isInitiater,
+    simplePeer: {
+      config: {
+        iceServers: [
```
Comment (WardMaes) [inline lib/p2p.ts:117]: You can use the config file here

File: machines/helper.ts [hunk 5]
```diff
@@ -3,12 +3,28 @@ import { MachineOptions, EventObject } from 'xstate'
...
+        ...config.delays,
+      }),
+      (masterConfig.guards = {
+        ...masterConfig.guards,
+        ...config.guards,
+      }),
+      (masterConfig.services = {
+        ...masterConfig.services,
+        ...config.services,
+      })
```
Comment (WardMaes) [inline machines/helper.ts:27]: Why all these parentheses?

Comment (WardMaes) [reply machines/helper.ts:27]: Not a huge fan of the mutations tho

File: lib/p2p.ts [hunk 6]
```diff
@@ -151,3 +103,35 @@ export async function sync(context: Context) {
...
 }
+
+const createPeer = (isInitiater: boolean, id?: string) => {
+  return new Peer({
+    id: id,
+    host: 'peer-connection.herokuapp.com',
+    secure: true,
+    port: 443,
+    path: '/peerjs/hive',
+    initiator: isInitiater,
```
Comment (WardMaes) [inline lib/p2p.ts:114]: `initiator`

File: lib/p2p.ts [hunk 7]
```diff
@@ -151,3 +103,35 @@ export async function sync(context: Context) {
 export const generateRoomId = () => {
   return Math.random().toString(26).substring(5, 10)
 }
+
+const createPeer = (isInitiater: boolean, id?: string) => {
```
Comment (WardMaes) [inline lib/p2p.ts:107]: Rename `isInitiater` to `initiator`

File: lib/p2p.ts [hunk 8]
```diff
@@ -151,3 +103,35 @@ export async function sync(context: Context) {
...
+  return new Peer({
+    id: id,
```
Comment (WardMaes) [inline lib/p2p.ts:109]: `id,`

**Review by WardMaes — COMMENTED** [review]
Lots of comments can be removed

**Comment by WardMaes** [comment]
@bestrandomnameever To recreate the bug: start game with 2 players, when both players have done 3 moves, the first player is unable to select a queen, but the second player is still able to click other insects in the `InsectSelector` and can thus place another insect, even if the queen hasn't been placed yet. 
The cursor shows the player is not able to click other insects (which is correct), but they are still clickable

File: components/InsectSelector.tsx (hunk 3 above)
Comment (bestrandomnameever) [reply components/InsectSelector.tsx:49]: I don't think I actually wrote this but moved it from somewhere (so it seems like I contributed it). 
image
Seems reasonable and with Cedric contribution, should be reduced to even smaller footprint. But agree that it's not with responsiveness in mind and could be improved. Let me know what you think.

File: machines/helper.ts (hunk 5 above)
Comment (bestrandomnameever) [reply machines/helper.ts:27]: Parentheses seem to be automatically added by formatter/prettier config. Can look into the rule that would allow this.

Comment (bestrandomnameever) [reply machines/helper.ts:27]: What do you mean by mutations? (the merging syntax?)

File: lib/p2p.ts (hunk 8 above)
Comment (bestrandomnameever) [reply lib/p2p.ts:109]: U wot makker?

Comment (WardMaes) [reply lib/p2p.ts:109]: `id: id,` is hetzelfde als `id,`

File: machines/helper.ts (hunk 5 above)
Comment (WardMaes) [reply machines/helper.ts:27]: `masterconfig` wordt hier 'gemuteerd', dat is niet zo mooi

File: components/InsectSelector.tsx (hunk 3 above)
Comment (WardMaes) [reply components/InsectSelector.tsx:49]: Ok, kan eventueel in andere PR opgelost worden

**Comment by bestrandomnameever** [comment]
@WardMaes Everything besides the fixed width insects in the insectPicker should now have been fixed.

**Comment by bestrandomnameever** [comment]
> @bestrandomnameever To recreate the bug: start game with 2 players, when both players have done 3 moves, the first player is unable to select anything other than his queen, but the second player is still able to click other insects in the `InsectSelector` and can thus place another insect, even if the queen hasn't been placed yet.
> The cursor shows the player is not able to click other insects (which is correct), but they are still clickable

This bug now fixed as well.

**Pull request merged**